Generation/src/components/GaussSmearVertex.cpp

```diff
@@ -0,0 +1,94 @@
+#include "GaussSmearVertex.h"
+
+#include "GaudiKernel/DeclareFactoryEntries.h"
+#include "GaudiKernel/IRndmGenSvc.h"
+#include "GaudiKernel/PhysicalConstants.h"
+#include "GaudiKernel/SystemOfUnits.h"
+#include "GaudiKernel/Vector4DTypes.h"
+
+/// Declaration of the Tool Factory
+DECLARE_TOOL_FACTORY(GaussSmearVertex)
+
+/// Standard constructor, initializes variables
+GaussSmearVertex::GaussSmearVertex(const std::string& type, const std::string& name, const IInterface* parent)
+    : GaudiTool(type, name, parent) {
+  declareInterface<IVertexSmearingTool>(this);
+
+  declareProperty("xVertexSigma", m_xsig = 0.0 * Gaudi::Units::mm);
+  declareProperty("yVertexSigma", m_ysig = 0.0 * Gaudi::Units::mm);
+  declareProperty("zVertexSigma", m_zsig = 0.0 * Gaudi::Units::mm);
```

Comment:
Why are these also declared as Properties if they're computed below from the other properties? Also move them into the header.